appmap/flask.py

```diff
@@ -45,62 +44,40 @@ def request_params(req):
     return values_dict(params.lists())
 
 
+NP_REGEXP = re.compile(r"<Rule '(.*?)'")
+NP_XLATE = re.compile(r"(?P<l><)|(?P<r>>)")
+
+
 class AppmapFlask(AppmapMiddleware):
-    def __init__(self, app=None):
+    """
+    A Flask extension to add remote recording to an application.
+    Should be loaded by default, but can also be added manually.
+
+    For example:
+
+    ```
+    from appmap.flask import AppmapFlask
+
+    app = new Flask(__Name__)
+    AppmapFlask().init_app(app)
+    ```
+    """
+
+    def __init__(self):
         super().__init__()
-        self.app = app
-        if app is not None:
-            self.init_app(app)
 
     def init_app(self, app):
         if self.should_record():
```

Comment:
The following check used to be in `init_app` and I had [removed](https://github.com/getappmap/appmap-python/pull/176/files#diff-144faf942a3ddaa37c23c0c4e9c67f3912c28aa7e267466a2250f4947be0431cL58) it.  Since `init_app` can now be called by the user, this check should probably be added back.

```
if not app:
    return
```

Maybe it should `raise` to let the user know app was `None`.

Reply:
It doesn't ever make sense to call `init_app` with `None`. Does it really add anything to check `app` and raise if it's `None`, rather than just allowing the runtime to do the check?

Reply:
Allowing the runtime to do the check seems better.